src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricEventSourceTests.cs

```diff
@@ -943,19 +987,31 @@ private void AssertInstrumentPublishingEventsPresent(EventWrittenEventArgs[] eve
             Assert.Equal(expectedInstruments.Length, publishEvents.Length);
         }
 
-        private void AssertCounterEventsPresent(EventWrittenEventArgs[] events, string meterName, string instrumentName, string tags,
+        private void AssertCounterRateEventsPresent(EventWrittenEventArgs[] events, string meterName, string instrumentName, string tags,
             string expectedUnit, params string[] expectedRates)
         {
-            AssertGenericCounterEventsPresent("CounterRateValuePublished", events, meterName, instrumentName, tags, expectedUnit, expectedRates);
+            AssertGenericCounterRateEventsPresent("CounterRateValuePublished", events, meterName, instrumentName, tags, expectedUnit, expectedRates);
```

Comment:
Rather than have duplicate AssertRate and AssertValue functions, perhaps it would be simpler to retain the original AssertCounterEvents function and change the string[] to (string,string)[] so that the tests can specify (rate,value) tuples?